### ISSUE TITLE
Error using flutter flavors in ios

### ISSUE THREAD

**Issue by imtoori** [flutter/flutter#19252]
## Steps to Reproduce

I set four schemes in my ios project in order to use flutter flavors. The problem is that my app shows a blank screen and logs some error that I can't understand.

I think the errors are these 
```
[ +924 ms] [VERBOSE-2:engine.cc(109)] Engine run configuration was invalid.
[   +2 ms] [VERBOSE-2:FlutterViewController.mm(170)] Could not launch engine with configuration.
```
## Logs

Output of `flutter run -t lib/main-dev.dart --flavor development --verbose` is attached
[log.txt](https://github.com/flutter/flutter/files/2183400/log.txt)

```
   info • Unused import: 'package:firebase_messaging/firebase_messaging.dart' • lib/app-state.dart:3:8
   info • Put @required named parameters first • lib/components/ensure-visible.dart:33:20
   info • Put @required named parameters first • lib/components/ensure-visible.dart:34:20
   info • Await future-returning functions inside async function bodies • lib/components/ensure-visible.dart:147:5
   info • Use => for short members whose body is a single return statement • lib/components/ensure-visible.dart:156:38
   info • Annotate overridden members • lib/components/notification-button.dart:8:13
   info • Don't override fields • lib/components/notification-button.dart:8:13
   info • Annotate overridden members • lib/models/maintenance.dart:7:16
   info • Annotate overridden members • lib/models/maintenance.dart:8:16
   info • Annotate overridden members • lib/models/maintenance.dart:9:16
   info • Annotate overridden members • lib/models/maintenance.dart:10:16
   info • Annotate overridden members • lib/models/maintenance.dart:11:16
   info • Annotate overridden members • lib/models/maintenance.dart:13:13
   info • Annotate overridden members • lib/models/maintenance.dart:15:13
   info • Annotate overridden members • lib/models/maintenance.dart:16:16
   info • Annotate overridden members • lib/models/maintenance.dart:17:16
   info • Annotate overridden members • lib/models/maintenance.dart:18:16
   info • Annotate overridden members • lib/models/maintenance.dart:19:16
   info • Annotate overridden members • lib/models/maintenance.dart:20:22
   info • Annotate overridden members • lib/models/notice.dart:7:13
   info • Annotate overridden members • lib/models/notice.dart:9:13
   info • Annotate overridden members • lib/models/notice.dart:10:16
   info • Annotate overridden members • lib/models/notice.dart:11:16
   info • Annotate overridden members • lib/models/notice.dart:12:16
   info • Annotate overridden members • lib/models/notice.dart:13:16
   info • Annotate overridden members • lib/models/notice.dart:14:18
   info • Annotate overridden members • lib/models/notice.dart:15:18
   info • Annotate overridden members • lib/models/place.dart:8:10
   info • Annotate overridden members • lib/models/place.dart:9:13
   info • Annotate overridden members • lib/models/place.dart:10:16
   info • Annotate overridden members • lib/models/place.dart:11:16
   info • Annotate overridden members • lib/models/place.dart:12:16
   info • Annotate overridden members • lib/models/place.dart:13:16
   info • Annotate overridden members • lib/models/place.dart:14:16
   info • Annotate overridden members • lib/models/place.dart:16:16
   info • Annotate overridden members • lib/models/place.dart:17:13
   info • Annotate overridden members • lib/models/place.dart:18:16
   info • Annotate overridden members • lib/models/place.dart:19:13
   info • Unused import: 'package:homeero_customer_app/api/api-manager-core.dart' • lib/pages/access-page.dart:6:8
   info • Unused import: 'package:homeero_customer_app/api/api-manager-core.dart' • lib/pages/email-feedback-page.dart:2:8
   info • Unused import: 'package:homeero_customer_app/app-config.dart' • lib/pages/maintenance-page.dart:3:8
   info • Only throw instances of classes extending either Exception or Error • lib/utils.dart:62:11

42 issues found. (ran in 5.4s)
```
```
[✓] Flutter (Channel beta, v0.5.1, on Mac OS X 10.13.5 17F77, locale it-IT)
    • Flutter version 0.5.1 at /Users/salvatore/Development/flutter
    • Framework revision c7ea3ca377 (6 weeks ago), 2018-05-29 21:07:33 +0200
    • Engine revision 1ed25ca7b7
    • Dart version 2.0.0-dev.58.0.flutter-f981f09760

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/salvatore/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4.1, Build version 9F2000
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 25.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[!] IntelliJ IDEA Community Edition (version 2017.2.6)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[!] VS Code (version 1.22.2)
    • VS Code at /Users/salvatore/Applications/Visual Studio Code.app/Contents
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected devices (1 available)
    • iPhone • 6a0400334b80ef69d8be2e5177290a9a0368a392 • ios • iOS 11.2.6
[log.txt](https://github.com/flutter/flutter/files/2183393/log.txt)

! Doctor found issues in 2 categories.
```

**Comment by zoechi** [comment]
Can you reproduce in `dev` channel as well?

**Comment by freewheelnat** [comment]
I have the same issue in channel dev v0.5.7.

Note: the app itself runs fine from XCode (selecting Product --> Run, with the Scheme set to my flavor)

FYI, I followed the [Flavoring Flutter](https://medium.com/@salvatoregiordanoo/flavoring-flutter-392aaa875f36) tutorial and I have a plist file for each flavor. However, with the --verbose tag on, I noticed that Info.plist was copied to device but not myflavor.plist. The difference between Info.plist and myflavor.plist is Bundle identifier and Bundle name. I have no idea if this is helpful, or just a red herring, I really don't know that much about iOS building an this is my first time trying flavors on iOS.
